mono/metadata/file-mmap-windows.c

```diff
@@ -385,12 +421,17 @@ mono_mmap_map (void *handle, gint64 offset, gint64 *size, int access, void **mma
 	// and size of the region of pages with matching attributes starting from base address.
 	// VirtualQueryEx: http://msdn.microsoft.com/en-us/library/windows/desktop/aa366907(v=vs.85).aspx
 	if (((viewInfo.State & MEM_RESERVE) != 0) || viewSize < (guint64) nativeSize) {
-		void *tempAddress = VirtualAlloc (address, nativeSize != 0 ? nativeSize : viewSize, MEM_COMMIT, get_page_access (access));
+		void *tempAddress;
+		MONO_ENTER_GC_SAFE;
+		tempAddress = VirtualAlloc (address, nativeSize != 0 ? nativeSize : viewSize, MEM_COMMIT, get_page_access (access));
+		MONO_EXIT_GC_SAFE;
 		if (!tempAddress) {
 			return convert_win32_error (GetLastError (), COULD_NOT_MAP_MEMORY);
 		}
```

Comment:
Consider `GetLastError` before `MONO_EXIT_GC_SAFE`. I need to check if `MONO_EXIT_GC_SAFE` can touch last error. Perhaps not, in which case, nevermind.

Reply:
It looks like it is ok. `MONO_EXIT_GC_SAFE` appears to deliberately preserve lasterror.

Reply:
Modulo a possible but in its first time initialization call..can be fixed separately.

Reply:
Are you planning to fix that bug yourself, or do I need to look at it? If the latter, can you point out exactly what the issue is?

Reply:
I'll fix it. It is a very rare or perhaps nonexistent thing. Maybe I misread the code.
Just for your curiousity, see around `threads_suspend_policy_getenv` and `W32_DEFINE_LAST_ERROR_RESTORE_POINT`.

probably best to deal with it in `hasenv_obsolete`.


Consider broadening `W32_DEFINE_LAST_ERROR_RESTORE_POINT` to also preserve `errno`, on all systems.

Reply:
Actually, fix it in the slow path of `mono_threads_suspend_policy` probably.

Reply:
Preserving `errno` would be pretty helpful, I didn't realize that wasn't the current behavior. I don't think it matters for this PR, but it's possible for the lack of that to have consequences elsewhere in the codebase.

Reply:
https://github.com/mono/mono/pull/14705 is what I was referring to.